google-cloud-firestore/src/main/java/com/google/cloud/firestore/QueryPlan.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.firestore;
+
+import java.util.Map;
+
+/** A QueryPlan contains information about the planning stage of a query. */
+public class QueryPlan {
```

Comment:
```suggestion
public final class QueryPlan {
```

I think pure data types can be made final. No need to extend of mock. Performance gain of being final.